e2e/test_operator_ha_upgrades/operator_ha_upgrade_test.go

```diff
@@ -434,4 +434,49 @@ var _ = Describe("Operator HA Upgrades", Label("e2e"), func() {
 		fixtures.GenerateUpgradeTableEntries(testOptions),
 	)
 
+	DescribeTable(
+		"upgrading a cluster when no remote processes are restarted",
+		func(beforeVersion string, targetVersion string) {
+			isAtLeast := factory.OperatorIsAtLeast(
+				"v1.14.0",
+			)
+
+			if !isAtLeast {
+				Skip("operator doesn't support feature for test case")
+			}
+
+			clusterSetup(beforeVersion, false)
+
+			// Select remote processes and use the buggify option to skip those
+			// processes during the restart command.
+			pods := fdbCluster.GetRemote().GetPods()
+			Expect(pods.Items).NotTo(BeEmpty())
+
+			ignoreDuringRestart := make(
+				[]fdbv1beta2.ProcessGroupID,
+				0,
+				len(pods.Items),
+			)
+
+			for _, pod := range pods.Items {
+				ignoreDuringRestart = append(
+					ignoreDuringRestart,
+					fdbv1beta2.ProcessGroupID(pod.Labels[fdbCluster.GetRemote().GetCachedCluster().GetProcessGroupIDLabel()]),
+				)
+			}
+
+			log.Println(
+				"Selected Pods:",
+				ignoreDuringRestart,
+				"to be skipped during the restart",
+			)
+			fdbCluster.GetRemote().SetIgnoreDuringRestart(ignoreDuringRestart)
```

Comment:
This is probably currently a limitation of the buggify setup for the ignore during restart, but I'm pretty sure this case will only work if the operator in the remote side is doing the restart, as the operator is currently iterating over the process groups from the cluster status (not the machine-readable status). We could change the logic here: https://github.com/FoundationDB/fdb-kubernetes-operator/blob/main/internal/buggify/buggify.go#L64-L72 to actually iterate over the machine-readable status, to make this feature multi-cluster aware.

We should also call the `SetIgnoreDuringRestart` on all clusters not only the remote side.